src/ai/backend/manager/models/vfolder.py

```diff
@@ -1383,40 +1385,67 @@ class Meta:
     status = graphene.String()
 
     @classmethod
-    def from_row(cls, ctx: GraphQueryContext, row: Row | VFolderRow) -> Optional[VirtualFolder]:
-        if row is None:
-            return None
-
-        def _get_field(name: str) -> Any:
-            try:
-                return row[name]
-            except sa.exc.NoSuchColumnError:
+    def from_row(cls, ctx: GraphQueryContext, row: Row | VFolderRow | None) -> Optional[Self]:
+        match row:
+            case None:
                 return None
-
-        return cls(
-            id=row["id"],
-            host=row["host"],
-            quota_scope_id=row["quota_scope_id"],
-            name=row["name"],
-            user=row["user"],
-            user_email=_get_field("users_email"),
-            group=row["group"],
-            group_name=_get_field("groups_name"),
-            creator=row["creator"],
-            domain_name=row["domain_name"],
-            unmanaged_path=row["unmanaged_path"],
-            usage_mode=row["usage_mode"],
-            permission=row["permission"],
-            ownership_type=row["ownership_type"],
-            max_files=row["max_files"],
-            max_size=row["max_size"],  # in MiB
-            created_at=row["created_at"],
-            last_used=row["last_used"],
-            # num_attached=row['num_attached'],
-            cloneable=row["cloneable"],
-            status=row["status"],
-            cur_size=row["cur_size"],
-        )
+            case VFolderRow():
+                return cls(
+                    id=row.id,
+                    host=row.host,
+                    quota_scope_id=row.quota_scope_id,
+                    name=row.name,
+                    user=row.user,
+                    user_email=row.user_row.email if row.user_row is not None else None,
+                    group=row.group,
+                    group_name=row.group_row.name if row.group_row is not None else None,
+                    creator=row.creator,
+                    domain_name=row.domain_name,
+                    unmanaged_path=row.unmanaged_path,
+                    usage_mode=row.usage_mode,
+                    permission=row.permission,
+                    ownership_type=row.ownership_type,
+                    max_files=row.max_files,
+                    max_size=row.max_size,  # in MiB
+                    created_at=row.created_at,
+                    last_used=row.last_used,
+                    cloneable=row.cloneable,
+                    status=row.status,
+                    cur_size=row.cur_size,
```

Comment:
The two codes are not the same although the meaning is almost identical.
For example, on the `Row` side, it is `row["id"]`, while on the `VFolderRow` side, it is `id=row.id`.
But I agree it would be great if there were a way to simplify the code further.